vladiate/vlad.py

```diff
@@ -88,31 +93,39 @@ def _log_missing(self, missing_items):
 
     def validate(self):
         self.logger.info(
-            "\nValidating {}(source={})".format(self.__class__.__name__, self.source)
+            "Validating {}(source={})".format(self.__class__.__name__, self.source)
         )
-        reader = csv.DictReader(self.source.open(), delimiter=self.delimiter)
 
-        if not reader.fieldnames:
-            self.logger.info(
-                "\033[1;33m" + "Source file has no field names" + "\033[0m"
+        if self.fieldnames:
+            reader = csv.DictReader(
+                self.source.open(), self.fieldnames, delimiter=self.delimiter
             )
+        else:
+            reader = csv.DictReader(self.source.open(), delimiter=self.delimiter)
+
+        if not reader.fieldnames:
+            self.logger.info("Source file has no field names.")
             return False
 
         self.missing_validators = set(reader.fieldnames) - set(self.validators)
         if self.missing_validators:
-            self.logger.info("\033[1;33m" + "Missing..." + "\033[0m")
+            if self.console_log:
+                self.logger.info("\033[1;33m" + "Missing..." + "\033[0m")
             self._log_missing_validators()
 
             if not self.ignore_missing_validators:
                 return False
 
         self.missing_fields = set(self.validators) - set(reader.fieldnames)
         if self.missing_fields:
-            self.logger.info("\033[1;33m" + "Missing..." + "\033[0m")
+            if self.console_log:
+                self.logger.info("\033[1;33m" + "Missing..." + "\033[0m")
             self._log_missing_fields()
             return False
 
         for line, row in enumerate(reader):
+            if line == 0 and self.fieldnames:
+                continue
```

Comment:
Can you explain this change?

Reply:
This check ensures that not to validate default csv header fieldnames against custom/override fieldnames. 
For example, `csv header fieldnames=['Column A', 'Column B'] and overide_fieldnames=['Column D', 'Column E']`
In this case, line=0, row={'Column D': 'Column A', 'Column E': 'Column B'} which is not actual data and should exclude from validation.